staging/src/k8s.io/apiserver/pkg/server/options/admission.go

```diff
@@ -94,3 +103,29 @@ func (a *AdmissionOptions) Validate() []error {
 	errs := []error{}
 	return errs
 }
+
+// preparePluginNamesIfNotProvided sets admission plugin names if they
+// were not provided by a cluster-admin. This method makes use of RecommendedPluginOrder
+// as well as DefaultOffPlugins fields.
+func (a *AdmissionOptions) preparePluginNamesIfNotProvided() {
+	if len(a.PluginNames) > 0 {
+		return
+	}
+	a.PluginNames = a.RecommendedPluginOrder
+
+	onlyEnabledPluginNames := []string{}
+	for _, pluginName := range a.PluginNames {
+		disablePlugin := false
+		for _, disabledPluginName := range a.DefaultOffPlugins {
+			if pluginName == disabledPluginName {
+				disablePlugin = true
+				break
+			}
+		}
+		if !disablePlugin {
+			onlyEnabledPluginNames = append(onlyEnabledPluginNames, pluginName)
+		}
+	}
+
+	a.PluginNames = onlyEnabledPluginNames
```

Comment:
I prefer code avoid side-effects.  How about `func (a *AdmissionOptions) getEnablePluginNames() string`?

Reply:
Sure, no problem. Just for clarification, in order to make `preparePluginNamesIfNotProvided` side-effect free we should not set `PluginNames` at all inside the method. Thus wouldn't be better to modify the method to `preparePluginNamesIfNotProvided []string` ?

Reply:
> Thus wouldn't be better to modify the method to preparePluginNamesIfNotProvided []string ?

Correct and where it's called, I'd have a local `enabledPlugins` variable.